test/puppetlabs/pcp/message_test.clj

```diff
@@ -102,43 +103,46 @@
              (vec (encode (set-data (make-message) (string->bytes "haha")))))))))
 
 (deftest decode-test
-  (with-redefs [schema.core/validate (fn [s d] d)]
-    (testing "it only handles version 1 messages"
-      (is (thrown+? [:type :puppetlabs.pcp.message/message-malformed]
-                    (decode (byte-array [2])))))
-    (testing "it insists on envelope chunk first"
-      (is (thrown+? [:type :puppetlabs.pcp.message/message-invalid]
-                    (decode (byte-array [1,
-                                         2, 0 0 0 2, 123 125])))))
-    (testing "it decodes the null message"
-      (is (= (dissoc (message->envelope (make-message)) :id)
-             (dissoc (message->envelope (decode (byte-array [1, 1, 0 0 0 2, 123 125]))) :id))))
-    (testing "it insists on a well-formed envelope"
-      (is (thrown+? [:type :puppetlabs.pcp.message/envelope-malformed]
-                    (decode (byte-array [1,
-                                         1, 0 0 0 1, 123])))))
-    (testing "it insists on a complete envelope"
-      (with-redefs [schema.core/validate (fn [s d] (throw (Exception. "oh dear")))]
-        (is (thrown+? [:type :puppetlabs.pcp.message/envelope-invalid]
-                      (decode (byte-array [1,
-                                           1, 0 0 0 2, 123 125]))))))
-    (testing "data is accessible"
-      (let [message (decode (byte-array [1,
-                                         1, 0 0 0 2, 123 125,
-                                         2, 0 0 0 3, 108 111 108]))]
-        (is (= (String. (get-data message)) "lol"))))
-    (testing "debug is accessible"
-      (let [message (decode (byte-array [1,
-                                         1, 0 0 0 2, 123 125,
-                                         2, 0 0 0 0,
-                                         3, 0 0 0 3, 108 111 108]))]
-        (is (= "lol" (String. (get-debug message))))))))
+  (testing "it only handles version 1 messages"
+    (is (thrown+? [:type :puppetlabs.pcp.message/message-malformed]
+                  (decode (byte-array [2])))))
+  (testing "it insists on envelope chunk first"
+    (is (thrown+? [:type :puppetlabs.pcp.message/message-invalid]
+                  (decode (byte-array [1,
+                                       2, 0 0 0 2, 123 125])))))
```

Comment:
I think it would be nice to have comments describing the message fields (version, envelope chunk - type, size, content, etc...), so that it would make sense to the reader.